renderdoc/core/android.cpp

```diff
@@ -708,6 +735,48 @@ bool PullAPK(const string &deviceID, const string &pkgPath, const string &apk)
   return false;
 }
 
+bool CheckLayerVersion(const string &deviceID, const string &layerName, const string &remoteLayer)
+{
+  RDCDEBUG("Checking layer version of: %s", layerName.c_str());
+
+  bool match = false;
+
+  // Use 'strings' command on the device to find the layer's build version
+  // i.e. strings -n <tag length> <layer> | grep <tag marker>
+  Process::ProcessResult result =
+      adbExecCommand(deviceID, "shell strings -n " +
+                                   StringFormat::ToString(strlen(RENDERDOC_Version_Tag_String) - 1) +
```

Comment:
Do you really need a new function `StringFormat::ToString` here? Could you not just use `StringFormat::Fmt` and a printf-style formatting string?

Reply:
Also here - it might be an idea to have a bit of a margin for error in the string-search length. E.g. consider if the version tag goes from `v1.9` to `v1.10`. I think searching for 2-3 shorter strings should do the trick.

Reply:
I'll give StringFormat::Fmt a try, didn't occur to me to check that.  to_string is just so convenient...  I'll add some wiggle room too, probably drop it down to 80.